rust/main/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -187,3 +192,189 @@ impl AppContextClassifier {
         Ok(None)
     }
 }
+
+/// An ISM caching policy.
+#[derive(Copy, Clone, Debug, PartialEq, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum IsmCachePolicy {
+    /// Default cache policy, includes the message in the cache key
+    /// when querying config that may be message-specific.
+    #[default]
+    MessageSpecific,
+    /// Even if an ISM's config interface is message-specific, we
+    /// ignore the message and use the same config for all messages.
+    IsmSpecific,
+}
+
+/// Configuration for ISM caching behavior.
+/// Fields are renamed to be all lowercase / without underscores to match
+/// the format expected by the settings parsing.
+#[derive(Debug, Clone, Default, Deserialize)]
+pub struct IsmCacheConfig {
+    #[serde(deserialize_with = "deserialize_module_types", rename = "moduletypes")]
+    module_types: HashSet<ModuleType>,
+    chains: Option<HashSet<String>>,
+    #[serde(default, rename = "cachepolicy")]
+    cache_policy: IsmCachePolicy,
+}
+
+/// To deserialize the module types from a list of numbers
+/// into a set of `ModuleType` enums.
+fn deserialize_module_types<'de, D>(deserializer: D) -> Result<HashSet<ModuleType>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let nums: Vec<u8> = Vec::deserialize(deserializer)?;
+    let mut set = HashSet::new();
+    for num in nums {
+        let module = ModuleType::from_u8(num).ok_or_else(|| {
+            serde::de::Error::custom(format!("Invalid module type value: {}", num))
+        })?;
+        set.insert(module);
+    }
+    Ok(set)
+}
+
+impl IsmCacheConfig {
+    fn matches_chain(&self, domain_name: &str) -> bool {
+        if let Some(chains) = &self.chains {
+            chains.contains(domain_name)
+        } else {
+            // If no domains are specified, match all domains
+            true
+        }
```

Comment:
nit: I thought that when `self.chains` was `None` we wouldn't cache anything. Seems like we always cache, and `self.chains` is used for deciding which chains to apply `cache_policy` to. To wasn't obvious to me but now I see the
```
        if root_ism == default_ism
            && self.default_ism_cache_config.matches_chain(domain.name())
            && self
                .default_ism_cache_config
                .matches_module_type(ism_module_type)
```
In that same if statement we could add a chain-specific flag to enable/disable caching, and have it off by default?

Reply:
added a config to just enable / disable globally